src/main/kotlin/org/rust/ide/annotator/RustHighlightingAnnotator.kt

```diff
@@ -2,73 +2,89 @@ package org.rust.ide.annotator
+                highlight(literal, RustHighlighter.map(literal.elementType))
             }
         }
 
-        override fun visitMacroInvocation(m: RustMacroInvocationElement) {
-            holder.highlight(m, RustColor.MACRO)
-        }
+        override fun visitTypeParam(o: RustTypeParamElement) = highlight(o.identifier, RustColor.TYPE_PARAMETER)
 
-        override fun visitTypeParam(o: RustTypeParamElement) {
-            holder.highlight(o, RustColor.TYPE_PARAMETER)
-        }
+        override fun visitAttr(o: RustAttrElement) = highlight(o, RustColor.ATTRIBUTE)
+
+        override fun visitTraitRef(o: RustTraitRefElement) = highlight(o.path.identifier, RustColor.TRAIT)
 
         override fun visitPatBinding(o: RustPatBindingElement) {
             if (o.isMut) {
-                holder.highlight(o.identifier, RustColor.MUT_BINDING)
+                highlight(o.identifier, RustColor.MUT_BINDING)
             }
         }
 
-        override fun visitPath(o: RustPathElement) {
-            o.reference.resolve().let {
-                if (it is RustPatBindingElement && it.isMut) {
-                    holder.highlight(o.identifier, RustColor.MUT_BINDING)
-                }
-            }
-        }
+        override fun visitEnumItem(o: RustEnumItemElement)       = highlight(o.identifier, RustColor.ENUM)
+        override fun visitEnumVariant(o: RustEnumVariantElement) = highlight(o.identifier, RustColor.ENUM_VARIANT)
 
-        override fun visitFnItem(o: RustFnItemElement) {
-            holder.highlight(o.identifier, RustColor.FUNCTION_DECLARATION)
-        }
+        override fun visitStructItem(o: RustStructItemElement)   = highlight(o.identifier, RustColor.STRUCT)
+        override fun visitTraitItem(o: RustTraitItemElement)     = highlight(o.identifier, RustColor.TRAIT)
+        override fun visitModDeclItem(o: RustModDeclItemElement) = highlight(o.identifier, RustColor.MODULE)
+        override fun visitModItem(o: RustModItemElement)         = highlight(o.identifier, RustColor.MODULE)
 
-        override fun visitImplMethodMember(o: RustImplMethodMemberElement) {
-            val color = if (o.isStatic) RustColor.STATIC_METHOD else RustColor.INSTANCE_METHOD
-            holder.highlight(o.identifier, color)
-        }
+        override fun visitFieldDecl(o: RustFieldDeclElement)     = highlight(o.identifier, RustColor.FIELD)
 
-        override fun visitTraitMethodMember(o: RustTraitMethodMemberElement) {
-            val color = if (o.isStatic) RustColor.STATIC_METHOD else RustColor.INSTANCE_METHOD
-            holder.highlight(o.identifier, color)
-        }
-    })
+        override fun visitExternCrateItem(o: RustExternCrateItemElement) = highlight(o.identifier, RustColor.CRATE)
 
-    private fun AnnotationHolder.highlight(element: PsiElement?, color: RustColor?) {
-        highlight(element, color?.textAttributesKey)
-    }
+        //TODO: When are element and element.identifier different?
```

Comment:
Macro invocation's production is:
```
elementType (".*_macro_invocation|macro_rules_invocation") = macro_invocation

...

// Invocations
macro_invocation ::= IDENTIFIER '!'
try_macro_invocation ::= "try" '!'
macro_rules_invocation ::= "macro_rules" '!'

// https://doc.rust-lang.org/std/fmt/
format_like_macro_invocation ::= ( "format"
                                 | "format_args"
                                 | "write"
                                 | "writeln"
                                 | "print"
                                 | "println"
                                 // logging macros from the log crate
                                 | "debug"
                                 | "error"
                                 | "info"
                                 | "log"
                                 | "trace"
                                 | "warn" ) '!'
```

So as you can see, `element.identifier` is the part without `!` here